turbopack/crates/turbo-rcstr/src/lib.rs

```diff
@@ -83,70 +134,73 @@ impl Deref for RcStr {
     type Target = str;
 
     fn deref(&self) -> &Self::Target {
-        self.0.as_str()
+        self.as_str()
     }
 }
 
 impl Borrow<str> for RcStr {
     fn borrow(&self) -> &str {
-        self.0.as_str()
+        self.as_str()
     }
 }
 
 impl From<Arc<String>> for RcStr {
     fn from(s: Arc<String>) -> Self {
-        RcStr(s)
+        match Arc::try_unwrap(s) {
+            Ok(v) => new_atom(Cow::Owned(v)),
+            Err(arc) => new_atom(Cow::Borrowed(&**arc)),
+        }
```

Comment:
Though I don't think this conversion is every really used, it should be possible to keep this as a fast-path by just converting the `Arc` to a pointer and adding the tag. (Assuming we keep the underlying `Arc<String>` representation, which doesn't seem to be currently happening?)

If we stick with `Arc<str>` for the internal representation, we should add a fast-path for that instead.

Reply:
I'll use `Arc<String>`. In this way we don't even need extra indirection via `Entry`.